sound/soc/sof/control.c

```diff
@@ -374,12 +384,13 @@ int snd_sof_bytes_ext_get(struct snd_kcontrol *kcontrol,
 	 */
 	size -= sizeof(const struct snd_ctl_tlv);
 
-	/* set the ABI header values */
-	cdata->data->magic = SOF_ABI_MAGIC;
-	cdata->data->abi = SOF_ABI_VERSION;
-
-	/* Prevent read of other kernel data or possibly corrupt response */
-	data_size = cdata->data->size + sizeof(const struct sof_abi_hdr);
+	/* check if this blob actually uses sof_abi_hdr */
+	if (cdata->data->magic == SOF_ABI_MAGIC)
+		data_size = cdata->data->size + sizeof(const struct sof_abi_hdr);
+	else {
+		dev_dbg(scomp->dev, "no sof_abi_hdr in blob\n");
+		data_size = cdata->data->size;
+	}
```

Comment:
this looks like a different patch?

Reply:
hmm, yes dealing with several PR's at the same time... sorry. should be updated now.